tests/test_serializers.py

```diff
@@ -1,75 +1,64 @@
 
-@override_settings(AEAT_CERT_PATH=None)
-@patch('aeat.rest_framework.serializers.aeat.Config', Mock())
-@patch('aeat.rest_framework.serializers.aeat.Controller')
-def test_make_aeat_request_without_cert_path_env_var_fails(ctrl_builder):
-    with pytest.raises(ImproperlyConfigured):
-        serializers.make_aeat_request('test_service', data={})
 
+def test_ens_serializer(zeep_response):
+    aeat_response = zeep_response(
+        'wsdl_ens_presentation_IE315V4.wsdl', 'ens_presentation_success_IE328V4Sal.xml', 'IE315V4'
+    )
 
-@override_settings(AEAT_KEY_PATH=None)
-@patch('aeat.rest_framework.serializers.aeat.Config', Mock())
-@patch('aeat.rest_framework.serializers.aeat.Controller')
-def test_make_aeat_request_without_key_path_env_var_fails(ctrl_builder):
-    with pytest.raises(ImproperlyConfigured):
-        serializers.make_aeat_request('test_service', data={})
+    serializer = serializers.ENSSerializer(data=aeat_response)
+    assert serializer.is_valid(raise_exception=False), serializer.errors
 
+    assert {'mrn': '17ES004311Z0000010'} == serializer.data
 
-@patch('aeat.rest_framework.serializers.make_aeat_request',
-       lambda x, y: Result(data=None, error='NO REENVIABLE'))
-def test_aeat_request_serializer_returns_result_failed_when_request_fails():
-    serializer = serializers.AEATRequest()
-    result = serializer.save()
-    assert not result.valid
 
+def test_exs_serializer(zeep_response):
+    aeat_response = zeep_response(
+        'wsdl_exs_IE615V2.wsdl', 'exs_presentation_success_IE628V2Sal.xml', 'IE615V2'
+    )
 
-@patch('aeat.rest_framework.serializers.make_aeat_request',
-       lambda x, y: Result(data='xyz', error=None, raw_response='xml...'))
-def test_aeat_request_serializer_returns_result_if_request_is_successfull():
-    serializer = serializers.AEATRequest()
-    result = serializer.save()
-    assert result.valid
+    serializer = serializers.EXSSerializer(data=aeat_response)
+    assert serializer.is_valid(raise_exception=False), serializer.errors
 
+    assert {'mrn': '17ES00361160001234',
+            'item_number_involved': 0, 'customs_intervention_code': 'V'} == serializer.data
 
-def test_adds_test_indicator_when_test_mode_is_enabled():
-    class TestSerializer(serializers.AEATRequest, rf.Serializer):
-        pass
 
-    serializer = TestSerializer(data={})
-    serializer.is_valid(raise_exception=True)
+@pytest.mark.parametrize('url,response,operation,expected,is_error,expected_data', [
+    # ENS
+    ('wsdl_ens_presentation_IE315V4.wsdl', 'ens_presentation_success_IE328V4Sal.xml', 'IE315V4',
+     serializers.ENSSerializer, False, {'mrn': '17ES004311Z0000010'}),
 
-    assert '0' == serializer.data['TesIndMES18']
+    ('wsdl_ens_presentation_IE315V4.wsdl', 'ens_presentation_error_IE316V4Sal.xml', 'IE315V4',
+     serializers.ENSFunctionalErrorSerializer, True,
+     {'type': '12', 'pointer': 'MES.MesSenMES3', 'reason': '1234-Message Sender is not valid'}),
```

Comment:
Functional errors **IE315V4** are handled too.
In errored response you should search for the _reason_ field.
Unknown responses has _reason_ field to